src/context.ts

```diff
@@ -69,9 +76,56 @@ function getContextFromWorkflow(): Context {
     }
   }
 
-  return context;
+  return {
+    commitDate: await getCommitDateFromWorkflow(context.sha, toolkit),
+    ...context
+  } as Context;
 }
 
 async function getContextFromGit(): Promise<Context> {
-  return await Git.context();
+  const ctx = await Git.context();
+
+  return {
+    commitDate: await Git.commitDate(ctx.sha),
+    ...ctx
+  } as Context;
+}
+
+async function getCommitDateFromWorkflow(sha: string, toolkit: Toolkit): Promise<Date> {
+  const event = GitHub.context.payload as unknown as {
+    // branch push
+    commits?: Array<{
+      timestamp: string;
+      // commit sha
+      id: string;
+    }>;
+    // tags
+    head_commit?: {
+      timestamp: string;
+      // commit sha
+      id: string;
+    };
+  };
+
+  if (event.commits) {
+    const commitDate = event.commits.find(x => x.id === sha)?.timestamp;
+    if (commitDate) {
+      return new Date(commitDate);
+    }
+  }
+
+  if (event.head_commit) {
+    if (event.head_commit.id === sha) {
+      return new Date(event.head_commit.timestamp);
+    }
+  }
+
+  // fallback to github api for commit date
+  const commit = await toolkit.github.octokit.request('GET /repos/{owner}/{repo}/commits/{commit_sha}', {
```

Comment:
I think you can just use `created_at` in `pull_request` event payload: https://github.com/docker/metadata-action/blob/32323e53fe6fa1b9830e2eed4bd13fd97d948ed6/__tests__/fixtures/event_pull_request.json#L193

Reply:
This is not the commit date, it's when PR is created

Reply:
for example

```text
  "title": "feat: add variable `commit_date`",
  "user": {...},
  "body": null,
  "created_at": "2024-11-13T13:01:45Z",
  "updated_at": "2024-11-14T07:10:48Z",
```